### PR TITLE
chore: Migrates`cloud_backup_snapshot_export_job` to new auto-generated SDK

### PR DESCRIPTION
**Pull request by EspenAlbert**
## Description

Migrates`cloud_backup_snapshot_export_job` to new auto-generated SDK

Link to any related issue(s): CLOUDP-246318

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

## Further comments

Passing acc+mig tests:

image

### CODE REVIEW

File: internal/service/cloudbackupsnapshotexportjob/data_source_cloud_backup_snapshot_export_job.go [hunk 1]
```diff
@@ -56,8 +56,9 @@ func DataSource() *schema.Resource {
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
 						"export_id": {
-							Type:     schema.TypeString,
-							Computed: true,
+							Type:       schema.TypeString,
```
Comment (EspenAlbert) [inline internal/service/cloudbackupsnapshotexportjob/data_source_cloud_backup_snapshot_export_job.go]: Waiting for reply from the Cloud Backup team on the status of this attribute

File: internal/service/cloudbackupsnapshotexportjob/resource_cloud_backup_snapshot_export_job.go [hunk 2]
```diff
@@ -162,61 +164,61 @@ func getRequiredFields(d *schema.ResourceData) (projectID, clusterName, exportID
...
 
-	if err := d.Set("created_at", exportJob.CreatedAt); err != nil {
+	if err := d.Set("created_at", conversion.TimePtrToStringPtr(exportJob.CreatedAt)); err != nil {
 		return diag.Errorf("error setting `created_at` for snapshot export job (%s): %s", d.Id(), err)
 	}
 
-	if err := d.Set("err_msg", exportJob.ErrMsg); err != nil {
-		return diag.Errorf("error setting `created_at` for snapshot export job (%s): %s", d.Id(), err)
-	}
+	// if err := d.Set("err_msg", exportJob.Err); err != nil {
```
Comment (EspenAlbert) [inline internal/service/cloudbackupsnapshotexportjob/resource_cloud_backup_snapshot_export_job.go]: Waiting for reply from the Cloud Backup team on the status of this attribute

File: internal/service/cloudbackupsnapshotexportjob/data_source_cloud_backup_snapshot_export_job.go [hunk 3]
```diff
@@ -56,8 +56,9 @@ func DataSource() *schema.Resource {
...
+							Computed:   true,
+							Deprecated: fmt.Sprintf(constant.DeprecationParamByVersion, "1.18.0"),
```
Comment (AgustinBettati) [inline internal/service/cloudbackupsnapshotexportjob/data_source_cloud_backup_snapshot_export_job.go]: would adjust docs if needed and also add a changelog entry mentioning the deprecation ([example](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/changelog-process.md#deprecations))

Comment (EspenAlbert) [reply internal/service/cloudbackupsnapshotexportjob/data_source_cloud_backup_snapshot_export_job.go]: Thanks. Added in 9187979

**Comment by AgustinBettati** [comment]
Pushed 2 temporary commits for adopting the latest sdk preview, these **must not** be included once PR transitions from draft.
Changes in cloud-provider-access and search-index are workarounds to adopt latest preview, but further investigation has to be done to understand these new breaking changes.

**Comment by oarbusi** [comment]
@EspenAlbert closing this PR in favor of https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2436

**Pull request closed without merging**